src/fun/mod.rs

```diff
@@ -78,7 +74,7 @@ pub enum Source {
   /// Was generated by the compiler.
   Generated,
   /// Source code location from the current book.
-  Local(Range<usize>),
+  Local(TextSpan),
```

Comment:
I wonder if we could and should unify all these source points and spans into a single thing. It's a bit overcomplicated atm.

Reply:
Maybe something like
```
struct Source {
  file: FileID/FileName,
  span: Range<usize>,
  kind: SourceKind,
}

enum SourceKind {
  User,
  Generated,
  Builtin
}
```
And then correctly propagate this information throughout the functions and terms, etc.

Reply:
Done in the latest commit